django/contrib/admin/options.py

```diff
@@ -216,7 +220,14 @@ def formfield_for_foreignkey(self, db_field, request, **kwargs):
         Get a form Field for a ForeignKey.
         """
         db = kwargs.get('using')
-        if db_field.name in self.raw_id_fields:
+
+        if db_field.name in self.autocomplete_fields:
+            related_admin = self.admin_site._registry.get(db_field.remote_field.model)
+            if not related_admin.search_fields:
+                msg = 'The admin for %s needs to implement "search_fields".'
```

Comment:
add: `"because it's referenced by <class>.autocomplete_fields."`
Maybe this should also be a system check ?